src/main/java/cubicchunks/worldgen/generator/flat/FlatLayerTypeGsonAdapter.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  This file is part of Cubic Chunks Mod, licensed under the MIT License (MIT).
+ *
+ *  Copyright (c) 2015 contributors
+ *
+ *  Permission is hereby granted, free of charge, to any person obtaining a copy
+ *  of this software and associated documentation files (the "Software"), to deal
+ *  in the Software without restriction, including without limitation the rights
+ *  to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ *  copies of the Software, and to permit persons to whom the Software is
+ *  furnished to do so, subject to the following conditions:
+ *
+ *  The above copyright notice and this permission notice shall be included in
+ *  all copies or substantial portions of the Software.
+ *
+ *  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ *  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ *  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ *  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ *  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ *  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ *  THE SOFTWARE.
+ */
+package cubicchunks.worldgen.generator.flat;
+
+import java.io.IOException;
+
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+
+import net.minecraft.block.state.IBlockState;
+import net.minecraftforge.fml.common.registry.GameData;
+
+public class FlatLayerTypeGsonAdapter extends TypeAdapter<Layer>{
+
+    @Override
+    public void write(JsonWriter out, Layer value) throws IOException {
+        out.beginObject();
+        out.name("fromY");
+        out.value(value.fromY);
+        out.name("toY");
+        out.value(value.toY);
+        out.name("blockState");
+        out.value(GameData.getBlockStateIDMap().get(value.blockState));
```

Comment:
It would be better to store string-based ID for bloc type, meta can be stored as number for now, but ideally it would be also stored as string (I know some commands can accept this kind of strings)